app/components/filters/date-range.js

```diff
@@ -0,0 +1,165 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { cached, tracked } from '@glimmer/tracking';
+
+export default class DateRangeFilter extends Component {
+  @tracked start;
+  @tracked end;
+  @tracked selectedPreset = null;
+  @tracked isChoosingPresets = true;
+  @tracked endDateError = [];
+
+  constructor(owner, args) {
+    super(owner, args);
+
+    const { start, end } = this.args;
+    this.start = start ? start : null;
+    this.end = end ? end : null;
+    this.findInitialDatePreset();
+  }
+
+  @cached
+  get presets() {
+    const year = new Date().getFullYear();
+    return [
+      {
+        label: `oktober-december ${year}`,
+        value: { start: `${year}-10-01`, end: `${year}-12-31` },
+      },
+      {
+        label: `juli-september ${year}`,
+        value: { start: `${year}-07-01`, end: `${year}-09-30` },
+      },
+      {
+        label: `april-juni ${year}`,
+        value: { start: `${year}-04-01`, end: `${year}-06-30` },
+      },
+      {
+        label: `januari-maart ${year}`,
+        value: { start: `${year}-01-01`, end: `${year}-03-31` },
+      },
+      {
+        label: `${year - 1}`,
+        value: { start: `${year - 1}-01-01`, end: `${year - 1}-12-31` },
+      },
+      {
+        label: `${year - 2}`,
+        value: { start: `${year - 2}-01-01`, end: `${year - 2}-12-31` },
+      },
+    ];
+  }
+
+  get hasBothDates() {
+    return Boolean(this.start) && Boolean(this.end);
+  }
+
+  get hasNoDates() {
+    return !this.start && !this.end;
+  }
+
+  get isInvalidDateRange() {
+    return this.hasBothDates && new Date(this.start) > new Date(this.end);
+  }
+
+  @action handleSelectionChange(selectedPreset) {
+    this.selectedPreset = selectedPreset;
+
+    if (selectedPreset) {
+      const { start, end } = selectedPreset.value;
+      this.start = start;
+      this.end = end;
+      this.triggerOnChange();
+    } else {
+      this.resetDates();
+    }
+  }
+
+  @action handleStartDateChange(newDate) {
+    this.start = newDate;
+
+    if (this.isDateComplete(newDate)) {
+      this.triggerOnChangeIfValid();
+    }
+  }
+
+  @action handleEndDateChange(newDate) {
+    this.end = newDate;
+
+    if (this.isDateComplete(newDate)) {
+      this.triggerOnChangeIfValid();
+    }
+  }
```

Comment:
At the moment we're triggering calling the onChange handler if either the start or end date changes. This is how it worked in OP. We could wait with triggering it until both are set (or unset) since we're only using the QP values if both are set.

I've kept it as is, since we might want to add a 'search from this date' or 'search until this date' feature in the future and the search handler already deals with only one of them being set.